pkg/store/instance.go

```diff
@@ -250,31 +264,96 @@ func PrintInstances(w io.Writer, instances []*Instance, format string) error {
 		}
 		fmt.Fprintln(w, string(b))
 	case "table":
+		types := map[string]int{}
+		archs := map[string]int{}
+		for _, instance := range instances {
+			types[instance.VMType]++
+			archs[instance.Arch]++
+		}
+		all := options != nil && options.AllFields
+		width := 0
+		if options != nil && options.IsTerminal {
+			width = options.TerminalWidth
+		}
+		hideType := false
+		hideArch := false
+		hideDir := false
+
+		columns := 1 // NAME
+		columns += 2 // STATUS
+		columns += 2 // SSH
+		if width != 0 && (columns+7)*8 > width && !all {
```

Comment:
Can we have a comment explaining where the magic `7` comes from? Maybe define a constant to give it a name.

Maybe also use a `columnWidth = 8` constant everywhere?

Reply:
The width of the remaining columns is somewhat inaccurate, because of the padding.

It probably has to calculate the width of all possibly combinations, and fit the biggest ?

All of these funky auto-hiding columns should probably have a template func or a subclass ?

```go
w := tabwriter.NewWriter(w, 4, 8, 4, ' ', 0)
```

Reply:
> The width of the remaining columns is somewhat inaccurate, because of the padding.
>
> It probably has to calculate the width of all possibly combinations, and fit the biggest ?

I would be fine with leaving the heuristics as-is, and maybe revisit this if or when we have examples where they fail dramatically.

I would just like the code to be a bit more self-explanatory. The column counting code is somewhat at odds with the hard-coded constants in the conditions.